packages/dds/merge-tree/src/test/resetPendingSegmentsToOp.spec.ts

```diff
@@ -3,9 +3,10 @@
  * Licensed under the MIT License.
  */
 
+/* eslint-disable unicorn/no-useless-spread */
```

Comment:
This rule has some false positives - we actually do need the spread operator, since the result of the mapping made by the autofix is an Iterable, not an array. 
https://github.com/sindresorhus/eslint-plugin-unicorn/issues/2018

Reply:
Let's leave a comment in the code explaining this next to the disable comment.

Reply:
We can do the same thing without the spread operator.
`const pending = [...client.mergeTree.pendingSegments.map((n) => n.data)];`
can be changed to 
`const pending = Array.from(client.mergeTree.pendingSegments.map((n) => n.data));`

Reply:
Or, since all we do is shift the array, we could replace that with just iterating and skip the array entirely?
```typescript
const pending = client.mergeTree.pendingSegments.map((n) => n.data);
const pendingIterator: Iterator<SegmentGroup, undefined> = pending[Symbol.iterator]();
opList = oldops.map((op) => ({
	op: client.regeneratePendingOp(op.op, pendingIterator.next().value!),
	refSeq: client.getCurrentSeq(),
}));
```

Reply:
In most such cases it probably complains at build time in which case it's not the *worst*, but I'm not sure that's always the case. IIRC @taylorsw04 or @noencke mentioned at some point getting toasted by linter autofixes in id-compressor that I think was also related to spread operators, but not sure if it was this rule or not.

Reply:
> FWIW, I don't really like having a rule in our recommended config with such a large gap like this. It has an autofixer which will make correct code incorrect (which was exactly what happened to Jillian here), presumably because it assumes the return of .map is already an array without actually having typing knowledge.

That seems like a good reason to disable the rule, yeah. Definitely feel free to disable in the config here for this PR, since it's clearly causing problems. I can get a PR up disabling it in our shared config.

Reply:
> The `Array.from` with 2 arguments seems good. I think that's a bit different from what I suggested to Jillian while she was looking at it, but a small variant on it was rejected by a 'prefer-spread' type rule.
> 
The Array.from is still getting rejected, so I'll leave the spread operator.

Reply:
PR disabling the rule in the base config: #21804